File: epoch_conversion/golang_epoch_conversion/epoch_conversion.go
package main

import "os"
import "fmt"
import "time"

func epoch_conversion_function(n int) int {
  return n
}

func date_utc() {
  utc := time.Now().UTC()
  year := utc.Year()
  month := utc.Month()
  day := utc.Day()
  hour := utc.Hour()
  min := utc.Minute()
  sec := utc.Second()
  fmt.Printf("local=<%d-%d-%d %d:%d:%d>\n", year, month, day, hour, min, sec)
  fmt.Println("utc=" + utc.String())
}

  func datetimeToEpoch( y int, m int, mday int, h int, min int, s int ) int {
    idx := 1970;
    total_days := 0;
    total_secs := 0;

    for ( idx < y ) {
      idx = idx + 1;
      total_days += compute_date_doy(idx, 12, 31);
    }
    total_days = total_days + compute_date_doy(y, m, mday - 1);
    total_secs = (total_days * 86400) + (h * 60 * 60) + (min * 60) + s;
    return total_secs;
  }

  func compute_date_doy( year int, month int, mday int ) int {
    n1 := (275 * month / 9);
    n2 := ((month + 9) / 12);
    n3 := (1 + ((year - 4 * (year / 4) + 2) / 3));
    n := n1 - (n2 * n3) + mday - 30;

    return n;
  }

func main() {
  args := os.Args
  argc := len(args)
  utc := time.Now().UTC()
  year := utc.Year()
  month1 := int(utc.Month())
  //month1 := 11
  day := utc.Day()
  hour := utc.Hour()
  min := utc.Minute()
  sec := utc.Second()


  if argc != 1 {
    fmt.Println(args[0] + " <noargs>")
    os.Exit(1)
  }

  fmt.Println(datetimeToEpoch(year, month1, day, hour, min, sec))
  date_utc();
}

File: epoch_conversion/ruby_epoch_conversion/epoch_conversion.rb
#!/usr/bin/ruby -w
#comments
require 'date'

def computeDateDoy( year, month, mday )
  n1 = ((275 * month) / 9).to_i
  n2 = ((month + 9) / 12).to_i
  n3 = (1 + ((year - 4 * (year / 4).to_i + 2) / 3).to_i)
  n = n1 - (n2 * n3) + mday - 30
  return n
end

def datetimeToEpoch ( years, months, mday, hours, mins, secs )
  epoch_start = 1970
  total_days = 0
  total_secs = 0

  while ( epoch_start < years )
    total_days = total_days + computeDateDoy(epoch_start, 12, 31)
    epoch_start = epoch_start + 1
  end
  total_days = total_days + computeDateDoy(years, months, mday - 1)
  total_secs = (total_days * 86400) + (hours * 60 * 60) + (mins * 60) + secs
  return total_secs
end

def main()
  utc = Time.now.utc
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  time = Time.now.to_i
  year = utc.year
  month = utc.month
  day = utc.day
  hour = utc.hour
  min = utc.min
  sec = utc.sec
  puts datetimeToEpoch(year,month,day,hour,min,sec)
  puts time
end

main()
